### ISSUE TITLE
padding-line-between-statements does not match async function

### ISSUE THREAD

**Issue by hotforfeature** [eslint/eslint#10121]
**Tell us about your environment**

* **ESLint Version:** 4.19.0
* **Node Version:** 8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": ["eslint:recommended"],
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "env": {
    "es6": true
  },
  "rules": {
    "padding-line-between-statements": [
      "error",
      { "blankLine": "always", "prev": "*", "next": "function" },
      { "blankLine": "always", "prev": "function", "next": "*" }
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

The `padding-line-between-statements` rule fails when using multiple keywords such as `async function` or `export function` when a rule is written for one of the keywords. This is because [the rule only looks at the first token](https://github.com/eslint/eslint/blob/master/lib/rules/padding-line-between-statements.js#L35), which doesn't match what the rule was written for.

```js
const a = true;
async function test() {}
function test2() {}
const b = true;
```

```bash
eslint --fix test.js
```

**What did you expect to happen?**

I would expect that the above configuration would require padding lines above and below *all* functions, including exported functions and async functions.

There are other use cases that this fails on. For example, any rules for "class" would fail with `export class`.

**What actually happened? Please include the actual, raw output from ESLint.**

Actual:

```js
const a = true;
async function test() {} // This function was skipped

function test2() {} // This function had the rule applied to it

const b = true;
```

Expected:

```js
const a = true;

async function test() {}

function test2() {}

const b = true;
```

**Comment by platinumazure** [comment]
Unrelated: Love the username!
